cas/service.c

```diff
@@ -556,6 +564,60 @@ m0_cas__ut_svc_be_get(struct m0_reqh_service *svc)
 	return service->c_be_domain;
 }
 
+static int cas_service_get_sdev_id(struct cas_service *cas_svc)
+{
+	struct m0_reqh         *reqh;
+	struct m0_conf_cache   *cache;
+	struct m0_conf_obj     *obj;
+	struct m0_fid          *cas_svc_fid;
+	struct m0_conf_service *service;
+	struct m0_conf_obj     *sdevs_dir;
+	struct m0_conf_sdev    *sdev = NULL;
+	bool                    found = false;
+	int                     rc = 0;
+
+	M0_ASSERT(cas_svc->c_sdev_id == UINT32_MAX);
+
+	if (cas_in_ut())
+		return M0_RC(0);
+
+	reqh = cas_svc->c_service.rs_reqh;
+	cache = &m0_reqh2confc(reqh)->cc_cache;
+
+	cas_svc_fid = &cas_svc->c_service.rs_service_fid;
+	obj = m0_conf_cache_lookup(cache, cas_svc_fid);
+	M0_ASSERT(obj != NULL);
+
+	service = M0_CONF_CAST(obj, m0_conf_service);
+	M0_ASSERT(service != NULL);
+
+	sdevs_dir = &service->cs_sdevs->cd_obj;
+	rc = m0_confc_open_sync(&sdevs_dir, sdevs_dir, M0_FID0);
+	if (rc != 0)
+		return M0_RC(rc);
+
+	obj = NULL;
+	while ((rc = m0_confc_readdir_sync(sdevs_dir, &obj)) > 0) {
+		sdev = M0_CONF_CAST(obj, m0_conf_sdev);
+		if (!found) {
+			cas_svc->c_sdev_id = sdev->sd_dev_idx;
+			found = true;
+		} else {
+			/*
+			 * Several devices attached to a single CAS service are
+			 * possible if we are run in ut. In this case we
+			 * do not replace the device ID and leave the ut alone.
+			 * Also we can not break the loop until whole directory
+			 * is iterated since it leads to crash during request
+			 * handler finalisation.
```

Comment:
Hmm... why? Do we understand what is the problem here?

Reply:
Not completely, need to root cause.